#45 Off-Screen-Arrange components

mliam0608 opened the pull request:
Created a new screen to represent off screen component.

Added an OffScreen.qml file to the content folder.

![image](https://github.com/Northeastern-Electric-Racing/Nero-2.0/assets/138416854/28bc5c12-d998-427e-9f6a-a88c93caf993)

Closes #45

mliam0608 committed:
#45 Off-Screen-Arrange components

Created a new screen to represent off screen component.

Peyton-McKee reviewed (changes requested):
all the components within this need to be based on the components height and width, right now everything is fixed which means if we ever want to rearrange things it would be a super pain in the ass

mliam0608 committed:
#45 Off Screen-Arrange Components

Changed layout so that components' positions are dependent on the screen size instead of being fixed.

mliam0608 commented:
![image](https://github.com/Northeastern-Electric-Racing/Nero-2.0/assets/138416854/2cf6fac0-98de-494f-b576-f952189b912b)

mliam0608 committed:
#45 Off screen - arrange components

Created gridLayouts for the title, values, and details. Each gridLayout can be adjusted individually.

mliam0608 committed:
#45 Off screen - arrange components

Changed the name to OffScreen.qml.

mliam0608 commented:
![image](https://github.com/Northeastern-Electric-Racing/Nero-2.0/assets/138416854/4a747490-3d44-40b6-94bb-2becb256562a)

Peyton-McKee commented on NERODesign/content/OffScreen.qml:
```diff
@@ -0,0 +1,303 @@
+import QtQuick.Layouts 1.15
+
+Item {
+    width: 1024
```
you should make this 800 by 480 to see what it would actually look like

Peyton-McKee commented on NERODesign/content/OffScreen.qml:
```diff
@@ -0,0 +1,303 @@
+            Layout.rowSpan: 1
+            width: parent.width * 0.45
+            height: parent.height * 0.1
+            shutdownFlowTask: "SIDE BRBs"
```
the design looks good, you also need to copy this to development and hook up the controller.

Peyton-McKee commented on NERODesign/content/OffScreen.qml:
```diff
@@ -0,0 +1,303 @@
+            Layout.column: 0
+            Layout.columnSpan: 1
+            Layout.rowSpan: 1
+            width: parent.width * 0.45
```
you should make this an actual value because we use it in a whole bunch of places

Peyton-McKee commented on NERODesign/content/OffScreen.qml:
```diff
@@ -0,0 +1,303 @@
+Item {
+    width: 1024
+    height: 768
+    visible: true
```
you also are going to have to link a selectedItem or something so that you can highlight the selected item.

Peyton-McKee reviewed (changes requested):
The design looks good, next step is connecting it to the controller in NERO development

mliam0608 committed:
#45 Off screen arrange components

Connect screen to controller.

mliam0608 committed:
Merge branch '#45-Arrange-components' of https://github.com/Northeastern-Electric-Racing/Nero-2.0 into #45-Arrange-components

mliam0608 committed:
#45 off screen arrange components

Added OffView to Contents.

mliam0608 committed:
#45 Off screen arrange components

Set size to 480 x 800 and make detailDisplay widths and heights constant.

Peyton-McKee committed:
Merge branch 'Develop' into #45-Arrange-components

Peyton-McKee committed:
#45 OffScreen Stable Design

Peyton-McKee committed:
#45 Modal Opening And Closing Working

Peyton-McKee commented:
https://github.com/Northeastern-Electric-Racing/Nero-2.0/assets/113635669/b9809045-bb56-4a47-93a9-86f0559ad6ee
